diff --git a/ckanext/dcat/harvesters/_json.py b/ckanext/dcat/harvesters/_json.py
index 6600ec8..1533f9d 100644
--- a/ckanext/dcat/harvesters/_json.py
+++ b/ckanext/dcat/harvesters/_json.py
@@ -168,11 +168,11 @@ class DCATJSONHarvester(DCATHarvester):
                 guid=guid, job=harvest_job,
                 package_id=guid_to_package_id[guid],
                 extras=[HarvestObjectExtra(key='status', value='delete')])
-            ids.append(obj.id)
             model.Session.query(HarvestObject).\
                 filter_by(guid=guid).\
                 update({'current': False}, False)
             obj.save()
+            ids.append(obj.id)
 
         return ids
